ledger/src/blockstore_db.rs

```diff
@@ -291,65 +291,33 @@ struct Rocks(rocksdb::DB, ActualAccessType, OldestSlot);
 
 impl Rocks {
     fn open(path: &Path, options: BlockstoreOptions) -> Result<Rocks> {
-        use columns::*;
-        let access_type = options.access_type;
-        let recovery_mode = options.recovery_mode;
+        let access_type = &options.access_type;
+        let recovery_mode = options.recovery_mode.clone();
 
         fs::create_dir_all(&path)?;
 
         // Use default database options
-        if matches!(access_type, AccessType::PrimaryOnlyForMaintenance) {
+        if disable_auto_compactions(access_type) {
```

Comment:
We had this in several places so created the common function `disable_auto_compations()`